common/types/map.go

```diff
@@ -83,17 +87,23 @@ func (m *stringMap) Contains(index ref.Val) ref.Val {
 }
 
 // ConvertToNative implements the ref.Val interface method.
-func (m *baseMap) ConvertToNative(refType reflect.Type) (interface{}, error) {
-	// JSON conversion.
-	if refType == jsonValueType || refType == jsonStructType {
+func (m *baseMap) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
+	switch typeDesc {
+	case anyValueType:
+		json, err := m.ConvertToNative(jsonValueType)
```

Comment:
I think g.p.Struct should be preferred here instead of the less-specific g.p.Value.